addons/mail/controllers/discuss/channel.py

```diff
@@ -47,15 +47,19 @@ def discuss_channel_avatar_update(self, channel_id, data):
 
     @http.route("/discuss/channel/messages", methods=["POST"], type="json", auth="public")
     def discuss_channel_messages(self, channel_id, before=None, after=None, limit=30, around=None):
-        channel_member_sudo = request.env["discuss.channel.member"]._get_as_sudo_from_request_or_raise(
-            request=request, channel_id=int(channel_id)
-        )
+        not_public_channel = request.env["discuss.channel"].sudo().browse(channel_id).group_public_id
```

Comment:
```suggestion
        channel = request.env["discuss.channel"].sudo().browse(channel_id)
        has_channel_access = not channel.group_public_id or channel.group_public_id in request.env.user.groups_id
```
This way it would work for channels which are not public but user has access to them.